modules/dxp/apps/scim/scim-configuration-web/src/main/java/com/liferay/scim/configuration/web/internal/portlet/action/SaveScimConfigurationMVCActionCommand.java

```diff
@@ -145,6 +145,17 @@ else if (Objects.equals(cmd, "revoke")) {
 			_oAuth2AuthorizationService.revokeAllOAuth2Authorizations(
 				oAuth2Application.getOAuth2ApplicationId());
 		}
+		else if (Objects.equals(cmd, "reset")) {
+			OAuth2Application oAuth2Application =
+				_oAuth2ApplicationLocalService.getOAuth2Application(
+					themeDisplay.getCompanyId(), scimClientId);
+
+			_oAuth2AuthorizationService.revokeAllOAuth2Authorizations(
+				oAuth2Application.getOAuth2ApplicationId());
+
+			_oAuth2ApplicationLocalService.deleteOAuth2Application(
+				oAuth2Application);
+		}
```

Comment:
The command `reset` comes before `revoke` alphabetically, so these conditions need reordering.  I will sort this myself, just pointing it out FYI.